integration/spark/app/src/test/java/io/openlineage/spark/agent/SparkStreamingTest.java

```diff
@@ -0,0 +1,280 @@
+          SparkSession.builder()
+              .appName("kafka-source-kafka-sink")
+              .master("local[*]")
+              .config("spark.extraListeners", OpenLineageSparkListener.class.getCanonicalName())
+              .config("spark.driver.host", "localhost")
+              .config("spark.driver.extraJavaOptions", "-Dderby.system.home=" + derbySystemHome)
+              .config("spark.sql.warehouse.dir", sparkSqlWarehouse.toString())
+              .config("spark.ui.enabled", false)
+              .config("spark.openlineage.transport.type", "http")
+              .config("spark.openlineage.transport.url", "http://localhost:" + randomPort)
+              .config("spark.openlineage.facets.disabled", "[spark_unknown;]")
+              .getOrCreate();
+
+      Dataset<Row> sourceStream =
+          spark
+              .readStream()
+              .format("kafka")
+              .option("subscribe", "source-topic")
+              .option("kafka.bootstrap.servers", bootstrapServers)
+              .option("startingOffsets", "earliest")
+              .load();
+
+      StructType schema = StructType.fromDDL("id STRING, epoch LONG");
+      StreamingQuery streamingQuery =
+          sourceStream
+              .selectExpr("CAST(value AS STRING) AS value")
+              .select(from_json(col("value"), schema).as("event"))
+              .select(col("event.id").as("id"), col("event.epoch").as("epoch"))
+              .select(col("id"), from_unixtime(col("epoch")).as("timestamp"))
+              .select(functions.struct(col("id"), col("timestamp")).as("converted_event"))
+              .select(functions.to_json(col("converted_event")).as("value"))
+              .writeStream()
+              .format("kafka")
+              .option("topic", "target-topic")
+              .option("kafka.bootstrap.servers", bootstrapServers)
+              .option("checkpointLocation", checkpointsDir.toString())
+              .trigger(Trigger.ProcessingTime(Duration.ofSeconds(4).toMillis()))
+              .start();
+
+      streamingQuery.awaitTermination(Duration.ofSeconds(20).toMillis());
+
+      spark.stop();
+
+      kafka.stop();
+
+      kafka.close();
+
+      Awaitility.await().atMost(Duration.ofSeconds(60)).until(() -> !kafka.isRunning());
+      Awaitility.await()
+          .atMost(Duration.ofSeconds(60))
+          .until(() -> spark.sparkContext().isStopped());
+
+      List<RunEvent> events =
+          handler.eventsContainer.stream()
+              .map(OpenLineageClientUtils::runEventFromJson)
+              .collect(Collectors.toList());
+
+      List<RunEvent> nonEmptyInputEvents =
+          events.stream().filter(x -> !x.getInputs().isEmpty()).collect(Collectors.toList());
+
+      assertEquals(6, nonEmptyInputEvents.size());
+
+      nonEmptyInputEvents.forEach(
+          event -> {
+            assertEquals(1, event.getInputs().size());
+            assertEquals("source-topic", event.getInputs().get(0).getName());
```

Comment:
Same as above - validate dataset schema. 
Would also be nice to validate a job type - should be STREAMING.